### PR TITLE
Fix not selectable last year

### LINKED ISSUE #1774: Last year is not clickable when use minDate and maxDate
## Environment

| Tech                 | Version |
| -------------------- | ------- |
| @material-ui/pickers |     4.0.0-alpha.7    |
| material-ui          |    4.9.10     |
| TypeScript           |         |
| React                |     16.13.1    |
| Browser              |         |
| Peer library         |    date-fns     |  |

## Steps to reproduce

Same code as mentioned in https://v4-0-0-alpha-7.material-ui-pickers.dev/demo/datepicker
  ```
     <DesktopDatePicker
        autoOk
        label="For desktop"
        renderInput={props => <TextField {...props} />}
        minDate={new Date("2017-06-06")}
        maxDate={new Date("2100-12-31")}
        value={selectedDate}
        onChange={date => handleDateChange(date)}
      />
```

When the minDate (2017-06-06) month is not starting Month i.e Jan, then when you click last year that is 2100, is not clickable
Screen Shot 2020-05-15 at 2 02 14 AM
## Expected behavior
If minDate does not have starting Month i.e. Jan then also last year should be clickable

## Actual behavior
Last Year 2100 is not clickable

**Can anyone provide the workaround or If possible can it be fix as hot-fix.** Thanks!

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
This PR help with #1774

### CODE REVIEW

File: lib/src/__tests__/DatePickerTestingLib.test.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        value={null}
+        onChange={jest.fn()}
+        views={['year']}
+        minDate="2000-01-01"
+        maxDate="2010-01-01"
+        renderInput={props => <TextField {...props} />}
+      />
+    );
+
+    expect(screen.getByText('2010').parentElement).toBeDisabled();
```
Comment (oliviertassinari) [inline lib/src/__tests__/DatePickerTestingLib.test.tsx]: `getByRole('button', { name: '2010' })` sounds like a more appropriate selector.

Comment (dmtrKovalenko) [reply lib/src/__tests__/DatePickerTestingLib.test.tsx]: Done

File: lib/src/__tests__/DatePickerTestingLib.test.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+import { DatePicker } from '../DatePicker';
+import { TextField } from '@material-ui/core';
+import { screen } from '@testing-library/react';
+import { createClientRender } from './createClientRender';
+
+describe('<DatePicker />', () => {
+  const render = createClientRender({ strict: false });
+  it('Allows to select edge years from list', () => {
```
Comment (oliviertassinari) [inline lib/src/__tests__/DatePickerTestingLib.test.tsx:9]:
```suggestion
  // StrictModeViolation: uses Grow
  const render = createClientRender({ strict: false });
  
  it('Should disable the year on the edge if it doesn`t fit to the current date', () => {
```

**Review by oliviertassinari — APPROVED** [review]

File: lib/src/__tests__/DatePickerTestingLib.test.tsx (hunk 2 above)
Comment (dmtrKovalenko) [reply lib/src/__tests__/DatePickerTestingLib.test.tsx:9]: Oh my gosh, just realize that this test will fail each year on January 1st. I hate writing tests for datepickers XD

**Pull request merged**